Chase_step1b_AssignOutcome_Algorithm3.sas

```diff
@@ -1,63 +1,90 @@
 
 
-********************************************************************;
-**** STEP 1b - Assigns outcome to each pregnancy outcome group  ****;
-**** Outcome Algorithm 3 Applied                                ****;
-********************************************************************;
 
-*modified from chase_1b3_assignoutcome_algorithm3.sas;
 /*
 *08.09 changed Not In to NE and remove Clean variables - instead apply Clean to outcome var;
 *      for CodeType capture type of code(s) present for outcome type (delivery/abortion/ectopic);
 *      (note alg directions separates rslt/cleaned+delivery/abortion outcome vars (outcome_b[2,4,5]);
 */
 
-*algorithm3 (complex) - checkng for delivery procedure, ignoring other non-delivery outcomes; 
-
 *** step 1b.2 discordnat flags ***;run;
 
+
+*First, deal with the pregnancy outcome groups with discordant outcomes.;
 data &discdsn._int_Alg3 ;
- set &discdsn (in=b);
-
-  Discordant3=0;
-
-  **summarize: ;
-  ** outcomes are delivery procedures or not;
-  **  if there is a delivery procedure then check for delivery concordance using 1b table1 (ignoring non-delivery outcomes);
-  **  if no deliver procedure flag as such and stop;
-
-  length Disc_OutPot3 $8.; 
-  *any delivery proc? check for concordant combos;;
-  if put(delivery,$outcdpr.) = 1 then do;;
-
-   **step 1b1 part 2;
-   *1 - ...and concordant delivery codes (redo table 1 check);
-    if lbm NE ('000') and udl NE ('000') And SB='000' And LBS='000' and MLS='000' 
-         then do; discordant3=1; Disc_OutPot3='PRD-LBM';end;
-    else if lbs NE ('000') and udl NE ('000') And SB='000' And LBM='000' and MLS='000' 
-        then do; discordant3=1; Disc_OutPot3='PRD-LBS';end;
-    else if mls NE ('000') and udl NE ('000') And SB='000' And LBM='000' and LBS='000'
-         then do; discordant3=1; Disc_OutPot3='PRD-MLS';end;
-    else if sb NE ('000') and udl NE ('000') And MLS='000' And LBM='000' and LBS='000' 
-        then do; discordant3=1; Disc_OutPot3='PRD-SB';end;
-
-    *note: this is somewhat redundant - only dif from above is UDL;
-    *if ignoring non-delivery outcomes (per note) then single delivery outcome types are concordent;
-    else if lbm NE ('000') and udl In ('000') And SB='000' And LBS='000' and MLS='000' 
-         then do; discordant3=1; Disc_OutPot3='PRD-LBM';end; *only delivery is lbm though may also have SAB etc;
-    else if lbs NE ('000') and udl In ('000') And SB='000' And LBM='000' and MLS='000' 
-        then do; discordant3=1; Disc_OutPot3='PRD-LBS';end;
-    else if mls NE ('000') and udl In ('000') And SB='000' And LBM='000' and LBS='000'
-         then do; discordant3=1; Disc_OutPot3='PRD-MLS';end;
-    else if sb NE ('000') and udl In ('000') And MLS='000' And LBM='000' and LBS='000' 
-        then do; discordant3=1; Disc_OutPot3='PRD-SB';end;
-    else if UDL NE ('000') and SB In ('000') And MLS='000' And LBM='000' and LBS='000' 
-        then do; discordant3=1; Disc_OutPot3='PRD-UDL';end;

```

Comment:
Simplified these into something simpler - Can eventually remove the commented out section.